clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -641,17 +641,15 @@ private void updateGroupMetadata(final Optional<Integer> memberEpoch, final Stri
 
     @Override
     public void registerMetricForSubscription(KafkaMetric metric) {
-        if (clientTelemetryReporter.isPresent()) {
-            ClientTelemetryReporter reporter = clientTelemetryReporter.get();
-            reporter.metricChange(metric);
+        if (!metrics().containsKey(metric.metricName())) {
+            clientTelemetryReporter.ifPresent(reporter -> reporter.metricChange(metric));
         }
     }
 
     @Override
     public void unregisterMetricFromSubscription(KafkaMetric metric) {
-        if (clientTelemetryReporter.isPresent()) {
-            ClientTelemetryReporter reporter = clientTelemetryReporter.get();
-            reporter.metricRemoval(metric);
+        if (!metrics().containsKey(metric.metricName())) {
+            clientTelemetryReporter.ifPresent(reporter -> reporter.metricRemoval(metric));
```

Comment:
Definitely don't want a way here to unregister an existing consumer metric from **_telemetry subscription_**

Reply:
So `metrics()` only returns user-registered metrics, but not of the consumer built-in metrics?

Reply:
`metrics()` returns the built-in consumer metrics - I probably wasn't clear enough in my comment above. The guard condition prevents from removing built-in consumer metrics